pkg/apis/db/v1alpha1/cassandracluster_types.go

```diff
@@ -89,29 +91,48 @@ const (
 	OperationRemove          string = "remove"
 )
 
-// SetDefaults sets the default values for the cassandra spec and returns true if the spec was changed
-func (cc *CassandraCluster) SetDefaults() bool {
-	changed := false
+// CheckDefaults chckes that required fields havent good values
+func (cc *CassandraCluster) CheckDefaults() {
 	ccs := &cc.Spec
-	if ccs.NodesPerRacks == 0 {
-		ccs.NodesPerRacks = 1
-		changed = true
-	}
+
 	if len(ccs.BaseImage) == 0 {
 		ccs.BaseImage = defaultBaseImage
-		changed = true
+	}
+	if len(ccs.Version) == 0 {
+		ccs.Version = defaultVersion
 	}
 	if len(ccs.ImagePullPolicy) == 0 {
 		ccs.ImagePullPolicy = defaultImagePullPolicy
-		changed = true
 	}
-	if len(ccs.Version) == 0 {
-		ccs.Version = defaultVersion
-		changed = true
+	if len(ccs.BootstrapImage) == 0 {
+		ccs.BootstrapImage = defaultBootstrapImage
+	}
+
+	//Init-Container 1 : init-config
+	if len(ccs.InitContainerImage) == 0 {
+		ccs.InitContainerImage = ccs.BaseImage + ":" + ccs.Version
 	}
+	if len(ccs.InitContainerCmd) == 0 {
+		ccs.InitContainerCmd = InitContainerCmd
+	}
+
 	if ccs.RunAsUser == nil {
 		ccs.RunAsUser = func(i int64) *int64 { return &i }(DefaultUserID)
 	}
+	if ccs.ReadOnlyRootFilesystem == nil {
+		ccs.ReadOnlyRootFilesystem = func(b bool) *bool { return &b }(true)
```

Comment:
Can't they be specified in the structure directly instead of setting them here if it's unset ?

Reply:
I've tried that in the past but with unsuccessful result :(